cmd/cluster/resizenode.go

```diff
@@ -0,0 +1,342 @@
+}
+
+// This command requires to be backplane logged into the target cluster
+// During command execution, the OCM context is swapped back and forth: cluster - hive - cluster
+func newCmdResizeNode(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags, globalOpts *globalflags.GlobalOptions) *cobra.Command {
+	ops := newResizeNodeOptions(streams, flags, globalOpts)
+	resizeNodeCmd := &cobra.Command{
+		Use:               "resizeNode",
+		Short:             "Resize a node. Requires previous login to the target cluster via ocm backplane login <target_cluster>",
+		Args:              cobra.NoArgs,
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ops.complete(cmd, args))
+			cmdutil.CheckErr(ops.run())
+		},
+	}
+	resizeNodeCmd.Flags().StringVar(&ops.node, "node", "", "The node to resize (e.g. ip-127.0.0.1.eu-west-2.compute.internal)")
+	resizeNodeCmd.Flags().StringVar(&ops.newMachineType, "machine-type", "", "The target AWS machine type to resize to (e.g. m4.xlarge)")
+	resizeNodeCmd.Flags().BoolVarP(&ops.verbose, "verbose", "", false, "Verbose output")
+	ops.k8sclusterresourcefactory.AttachCobraCliFlags(resizeNodeCmd)
+	resizeNodeCmd.MarkFlagRequired("cluster-id")
+	resizeNodeCmd.MarkFlagRequired("node")
+	resizeNodeCmd.MarkFlagRequired("machine-type")
+
+	return resizeNodeCmd
+}
+
+func newResizeNodeOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags, globalOpts *globalflags.GlobalOptions) *resizeNodeOptions {
+	return &resizeNodeOptions{
+		k8sclusterresourcefactory: k8spkg.ClusterResourceFactoryOptions{
+			Flags: flags,
+		},
+		IOStreams:     streams,
+		GlobalOptions: globalOpts,
+	}
+}
+
+func (o *resizeNodeOptions) complete(cmd *cobra.Command, _ []string) error {
+	err := CompleteValidation(&o.k8sclusterresourcefactory, o.IOStreams)
+	if err != nil {
+		return err
+	}
+
+	utils.IsValidClusterKey(o.k8sclusterresourcefactory.ClusterID)
+
+	describedCluster, err := utils.DescribeCluster(o.k8sclusterresourcefactory.ClusterID)
+	if err != nil {
+		return err
+	}
+
+	if strings.ToUpper(describedCluster.CloudProvider().ID()) != "AWS" {
+		return fmt.Errorf("This command is only available for AWS clusters")
+	}
+	/*
+		Ideally we would want additional validation here for:
+		- the machine type exists
+		- the node exists on the cluster
+		- this command isn't used on stage
+
+		As this command is idempotent, it will just fail on a later stage if e.g. the
+		machine type doesn't exist and can be re-run.
+	*/
+
+	return nil
+}
+
+func (o *resizeNodeOptions) initAwsCli() *awsprovider.Client {
+	fmt.Println("Attempting to initialize AWS Client. Switching to hive context to get credentials.")
+
+	targetCluster := o.k8sclusterresourcefactory.ClusterID
+	hiveCluster, err := utils.GetHiveCluster(targetCluster)
+	if err != nil {
+		log.Fatalln("Unable to get hive cluster:", err)
+	}
+
+	utils.SwapOCMContext(hiveCluster)
```

Comment:
Shouldn't you capture the error here with `err =`?

Reply:
Good catch, thank you :)